tools/travis_before_install.sh

```diff
@@ -28,33 +28,49 @@ retry () {
 }
 
 
-# on Python 2.7, use the system versions of numpy, scipy, and matplotlib
-# and the minimum version of cython and networkx
-if [[ $TRAVIS_PYTHON_VERSION == 2.7* ]]; then
+# on Python 2.7, use the system versions of matplotlib, numpy and scipy
+# which is also our minimum requirements
+# this will change if travis updates to ubuntu LTS 14.04
+# http://docs.travis-ci.com/user/ci-environment/#CI-environment-OS
+if [ $TRAVIS_PYTHON_VERSION == 2.7 ]; then
+    sudo apt-get install python-scipy python-matplotlib python-numpy python-imaging
+    # above should be equal to minimum requirements
+    # remove from requirements and set the rest to minimum version
+    sed -i '/^matplotlib.*$/d' requirements.txt
+    sed -i '/^scipy.*$/d' requirements.txt
+    sed -i '/^numpy.*$/d' requirements.txt
+    sed -i '/^pillow.*$/d' requirements.txt
+    sed -i 's/>=/==/g' requirements.txt
+    # remove pillow from wheel install
+    export WHEELBINARIES=${WHEELBINARIES/pillow/}
+    # use system python
     virtualenv --system-site-packages ~/venv
```

Comment:
@blink1073 I'm only running this 2.7 test of now, not using `-p` there. I [did get past](https://travis-ci.org/scikit-image/scikit-image/jobs/63893611) this `pkg_resource` error by installing `python-pip` and `python-setuptools` earlier on. But then it failed on `travis_script.sh`, trying to install the optional dependencies with pip.

Debugging the issue further showed that pip in `travis_script.sh` [was version 1.0](https://travis-ci.org/scikit-image/scikit-image/jobs/63893611#L3174), even if [pip was upgraded](https://travis-ci.org/scikit-image/scikit-image/jobs/63893611#L433) and [verified as version 7.0.1](https://travis-ci.org/scikit-image/scikit-image/jobs/63893611#L568).

So the issue that exists seem to be:
1. [System pkg_resources](https://travis-ci.org/scikit-image/scikit-image/jobs/63893611#L4981) is used, creating trouble for setuptools.
2. Something happens with the $PATH when jumping to the next script.

Also, here's the virtualenv manual on `-p`:
>  -p PYTHON_EXE, --python=PYTHON_EXE
>                        The Python interpreter to use, e.g.,
>                        --python=python2.5 will use the python2.5 interpreter
>                        to create the new environment.  The default is the
>                        interpreter that virtualenv was installed with
>                        (/usr/bin/python)

If I understand that correctly, one can be bitten if `virtualenv` was installed by another python intepreter. Not sure if that is possible, and we are not bitten by it: [travis 3.3 venv](https://travis-ci.org/scikit-image/scikit-image/jobs/63883429#L97), [our venv](https://travis-ci.org/scikit-image/scikit-image/jobs/63883429#L1245).